src/main/groovy/org/spongepowered/asm/gradle/plugins/MixinEclipse.groovy

```diff
@@ -53,6 +53,22 @@ public class MixinEclipse {
             return
         }
         
+        def eclipseAptPlugin = project.plugins.findPlugin('com.diffplug.eclipse.apt')
+        if (eclipseAptPlugin) {
+            def settings = project.tasks.register('mixinEclipseJdtApt', EclipseJdtAptTask.class) {
+                description = 'Creates the Eclipse JDT APT settings file'
+                output = project.file('.settings/org.eclipse.jdt.apt.core.prefs')
+                mappingsIn = extension.mappings
+                dependsOn project.tasks.createSrgToMcp
+                dependsOn project.tasks.createMcpToSrg
+                dependsOn project.tasks.eclipseJdtApt
+                hasDiffplug = true
+            }
+            project.tasks.eclipse.dependsOn(settings)
+            return
+        }
+        project.logger.warn '[MixinGradle] Use \'com.diffplug.eclipse.apt\' gradle plugin for full annotation support within eclipse'
```

Comment:
This should probably be in an else block so that it only logs if the apt plugin is not applied.

Additionally, does this mean users must apply 2 eclipse-related plugins?

Reply:
1) Probably, yeah

2) No, because it isn't strictly required - only if you need the AP support within eclipse - if there is a way we could make this plugin dep on diffplug and automatically pull it, that would be the best solution imo